store-implementation/proxy-store/src/main/java/uk/gov/gchq/gaffer/proxystore/ProxyStore.java

```diff
@@ -0,0 +1,305 @@
+            throw new OperationException("Unable to serialise operation chain into JSON.", e);
+        }
+
+        final URL url = getProperties().getGafferUrl("graph/doOperation");
+        try {
+            return doPost(url, opChainJson, operationChain.getOutputTypeReference(), context);
+        } catch (final StoreException e) {
+            throw new OperationException(e.getMessage(), e);
+        }
+    }
+
+    protected <OUTPUT> OUTPUT doPost(final URL url, final Object body,
+                                     final TypeReference<OUTPUT> outputType) throws StoreException {
+        return doPost(url, body, outputType, null);
+    }
+
+    protected <OUTPUT> OUTPUT doPost(final URL url, final Object body,
+                                     final TypeReference<OUTPUT> outputType,
+                                     final Context context) throws StoreException {
+        try {
+            return doPost(url, new String(JSON_SERIALISER.serialise(body), CommonConstants.UTF_8), outputType, context);
+        } catch (SerialisationException | UnsupportedEncodingException e) {
+            throw new StoreException("Unable to serialise body of request into json.");
+        }
+    }
+
+    protected <OUTPUT> OUTPUT doPost(final URL url, final String jsonBody,
+                                     final TypeReference<OUTPUT> clazz,
+                                     final Context context) throws StoreException {
+
+
+        final Builder request = createRequest(jsonBody, url, context);
+        final Response response;
+        try {
+            response = request.post(Entity.json(jsonBody));
+        } catch (final Exception e) {
+            throw new StoreException("Failed to execute post via " +
+                    "the Gaffer URL " + url.toExternalForm(), e);
+        }
+
+        return handleResponse(response, clazz);
+    }
+
+    protected <OUTPUT> OUTPUT doGet(final URL url,
+                                    final TypeReference<OUTPUT> outputTypeReference)
+            throws StoreException {
+        final Invocation.Builder request = createRequest(null, url, null);
+        final Response response;
+        try {
+            response = request.get();
+        } catch (final Exception e) {
+            throw new StoreException("Request failed to execute via url "
+                    + url.toExternalForm(), e);
+        }
+
+        return handleResponse(response, outputTypeReference);
+    }
+
+    protected <OUTPUT> OUTPUT handleResponse(final Response response,
+                                             final TypeReference<OUTPUT> outputTypeReference)
+            throws StoreException {
+        String outputJson = null;
+        switch (response.getStatus()) {
+            case 200:
+                outputJson = response.readEntity(String.class);
+                break;
+            case 204:
+                break;
+            default:
```

Comment:
I'm not sure. We could throw a different exception or provide some contextual information about why it was a 400/500. What do you think?